src/ray/raylet/node_manager.cc

```diff
@@ -507,6 +507,89 @@ void NodeManager::DoLocalGC() {
   }
 }
 
+void NodeManager::SpillObjects(const std::vector<ObjectID> &objects_ids_to_spill,
+                               std::function<void(const ray::Status &)> callback) {
+  std::vector<ObjectID> objects_ids;
+  for (const auto &id : objects_ids_to_spill) {
+    // Do not spill already spilled objects.
+    if (spilled_objects_.count(id) == 0) {
+      objects_ids.push_back(id);
+    }
+  }
+  if (objects_ids.empty()) {
+    if (callback) {
+      callback(Status::OK());
+    }
+    return;
+  }
+  worker_pool_.PopIOWorker(
+      [this, objects_ids, callback](std::shared_ptr<WorkerInterface> io_worker) {
+        RAY_LOG(DEBUG) << "Sending object spilling request";
+        rpc::SpillObjectsRequest request;
+        for (const auto &object_id : objects_ids) {
+          request.add_object_ids_to_spill(object_id.Binary());
+        }
+        io_worker->rpc_client()->SpillObjects(
+            request, [this, objects_ids, callback, io_worker](
+                         const ray::Status &status, const rpc::SpillObjectsReply &r) {
+              worker_pool_.PushIOWorker(io_worker);
+              if (!status.ok()) {
+                RAY_LOG(ERROR) << "Failed to send object spilling request: "
+                               << status.ToString();
+              } else {
+                RAY_CHECK(static_cast<size_t>(r.spilled_objects_url_size()) ==
+                          objects_ids.size());
+                for (size_t i = 0; i < objects_ids.size(); ++i) {
+                  const ObjectID &object_id = objects_ids[i];
+                  const std::string &object_url = r.spilled_objects_url(i);
+                  // TODO(suquark): write to object directory.
+                  spilled_objects_[object_id] = object_url;
```

Comment:
If the same object gets force-spilled again before the first spilling is complete, could there be a problem here since the entry won't be in spilled_objects_?

Reply:
right, this is a problem, but I prefer solving this problem later for automatic object spilling (because we would not expect calling `force_spill_objects` on a same object multiple times anyway).